### PR TITLE
Text editor tabs + Commenting refactor 

### PR DESCRIPTION
**Pull request by puopg**

### CODE REVIEW

**Comment by garronmichael** [comment]
Good job!

File: client/app/room/room.html [hunk 1]
```diff
@@ -1,8 +1,18 @@
 <div id="room-container">
     <h1>Room</h1>
-    <button type="buton"ng-click="saveData()">Save Data</button>
```
Comment (joshuaNewman10) [inline client/app/room/room.html]: Whyd we get rid of the saveData button?

File: client/app/room/roomCtrl.js [hunk 2]
```diff
@@ -8,41 +11,62 @@
     .module('hackbox')
     .controller('roomCtrl', RoomCtrl);
 
+  RoomCtrl.$inject = ['$scope', '$sce', 'Video', 'Drawing', 'TextEditor'];
 
-  RoomCtrl.$inject = ['$scope', '$sce', 'Video', 'Drawing', '$http'];
-
-  function RoomCtrl($scope, $sce, Video, Drawing, $http){
+  function RoomCtrl($scope, $sce, Video, Drawing, TextEditor){
```
Comment (joshuaNewman10) [inline client/app/room/roomCtrl.js]: Are we intending on refactoring the saving functionality into another controller? If not the room controller needs the http dependency to save data

File: client/app/room/roomCtrl.js [hunk 3]
```diff
@@ -8,41 +11,62 @@
...
+     *
+     * @param roomName: A string representing the roomname to join. 
+     */
     $scope.initializeVideo = function(roomName){
       // Create the Icecomm object and get the instance of it.
       var comm = Video.getIcecommInstance();
 
       // Connect to the correct room.
-      comm.connect(roomName, {audio: false});
+      comm.connect(roomName, {limit: 2, audio: false});
```
Comment (joshuaNewman10) [inline client/app/room/roomCtrl.js]: Would be great to have a comment here saying that limit sets the max number of connections in a room

File: client/app/room/roomCtrl.js [hunk 4]
```diff
@@ -61,11 +85,12 @@
...
-    //it then appends the canvas to the DOM and wraps
-    //the canvas element inside Fabric
-    //lastly it gives a reference of the fabric object to the room controller's scope
+    /**
+     * Function: RoomCtrl.initializeCanvas(containerClassName)
+     * This function will append a canvas element to the room. 
+     *
+     * @param containerClassName: NOT_USED_CURRENTLY
+     */
     $scope.initializeCanvas = function(containerClassName) {
```
Comment (joshuaNewman10) [inline client/app/room/roomCtrl.js]: containerClassName has been removed. It will show the the update in my next pull request

**Pull request merged**